tools/cli/src/config.rs

```diff
@@ -2,20 +2,83 @@ use anyhow::{Context, Result};
 use layer_climb::prelude::*;
 use serde::{Deserialize, Serialize};
 
+// This is the on-disk config
+// it's not the final config that gets filled in asynchrnously
+#[derive(Debug, Deserialize, Serialize)]
+pub struct OnDiskConfig {
+    pub chains: ChainConfigs,
+    pub faucet: FaucetConfig,
+    pub wasmatic: OnDiskWasmaticConfig,
+}
+#[derive(Debug, Deserialize, Serialize)]
+pub struct OnDiskWasmaticConfig {
+    pub endpoint: String,
+}
+
+// This is the actual, final config that gets used throughout the app
 #[derive(Debug, Deserialize, Serialize)]
 pub struct Config {
     pub chains: ChainConfigs,
     pub faucet: FaucetConfig,
+    pub wasmatic: WasmaticConfig,
 }
 
 impl Config {
     // Load the config from the file
     // but in theory this could be from chain, http endpoint, avs, etc.
+    // internally, it does additional loads as needed (e.g. from wasmatic endpoint)
     pub async fn load() -> Result<Self> {
-        serde_json::from_str(include_str!("../config.json")).context("Failed to parse config")
+        let config: OnDiskConfig = serde_json::from_str(include_str!("../config.json"))
+            .context("Failed to parse config")?;
+
+        // SANITY CHECK
+        // make sure every chain has the same address kind
+
+        let address_kind = match (&config.chains.local, &config.chains.testnet) {
+            (Some(local), Some(testnet)) => {
+                if local.address_kind != testnet.address_kind {
+                    return Err(anyhow::anyhow!(
+                        "Local and testnet chains must have the same address kind"
+                    ));
+                }
+
+                &local.address_kind
+            }
+            (Some(local), None) => &local.address_kind,
+            (None, Some(testnet)) => &testnet.address_kind,
+            (None, None) => return Err(anyhow::anyhow!("At least one chain must be configured")),
+        };
+
+        let wasmatic_address = load_wasmatic_address(&config.wasmatic.endpoint).await?;
+
+        Ok(Config {
+            wasmatic: WasmaticConfig {
+                endpoint: config.wasmatic.endpoint,
+                address: match &address_kind {
+                    // TODO- this should be a method on AddrKind
+                    AddrKind::Cosmos { prefix } => {
+                        Address::new_cosmos_string(&wasmatic_address, Some(prefix))?
+                    }
+                    AddrKind::Eth => Address::new_eth_string(&wasmatic_address)?,
+                },
+            },
+            chains: config.chains,
+            faucet: config.faucet,
+        })
     }
 }
 
+async fn load_wasmatic_address(_endpoint: &str) -> Result<String> {
+    // TODO: Load from endpoint
```

Comment:
Ah, the key point here.
Note, in the future, we may want some Vec (and Vec of endpoints), but I'll make another issue for that